analysis/task_analysis_surface/second_level_analysis_surf_ss.py

```diff
@@ -0,0 +1,149 @@
+import sys
+import os
+import pandas as pd
+import numpy as np
+import glob
+import json
+
+import nilearn
+import nibabel as nb
+
+from nilearn.glm.contrasts import _compute_fixed_effects_params
+from nilearn import image as nimg
+
+
+#function to run second level fixed effects model to combine runs
+
+def run_second_level_model(sub,task,ses):
+    memory = 'nilearn_cache' #change if desired
+    
+    #find all contrasts for the two runs
+    effect_size_list_all = glob.glob(f'../../../derivatives/task_analysis_surface/first_level/sub-{sub}/ses-{ses}/task-{task}/sub-{sub}_ses-{ses}_task-{task}_rec-unco_run-*_contrast-*_effect_size.dscalar.nii')
+    
+    #create contrast to effect size path directory
+    effect_size_dict = {}
+    for path in effect_size_list_all:
+        contrast = path.split('contrast-')[-1].split('_effect')[0]
+        if contrast not in effect_size_dict.keys():
+            effect_size_dict[contrast]=[]
+        effect_size_dict[contrast].append(path)
+        
+    #do fixed effects analysis of all available contrasts 
+    output_dict={}
+    for contrast in effect_size_dict.keys():
+        effect_size_list = effect_size_dict[contrast]
+        #check if it was possible to calculate a given contrast for both runs
+        #if not, skip this contrast
+        if len(effect_size_list) < 2:
+            print(f'Only one {contrast} contrast for {sub} during {task} in {ses}. Second level analysis not possible.')
+            continue
+        effect_variance_list=glob.glob(f'../../../derivatives/task_analysis_surface/first_level/sub-{sub}/ses-{ses}/task-{task}/sub-{sub}_ses-{ses}_task-{task}_rec-unco_run-*_contrast-*_variance.dscalar.nii')
+        
+        #TO DO: figure out meaning of precision weighted
+        #note that results are made up of these: fx_results = [fixed_fx_contrast, fixed_fx_variance, fixed_fx_tstat]
+        fx_results = _compute_fixed_effects_params(
```

Comment:
does nilearn really require a hidden function to compute this? is there no public api for this?

Reply:
or is this because of cifti?

Reply:
yes it's because of cifti. that's the only time when i went to hidden functions